platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,1026 @@
+
+1. `TssMessageMap` - A map of TssMessageTransactions that have come to consensus for a roster.
+2. `TssVoteMap` - A map of TssVoteTransactions that have come to consensus for a roster.
+
+Keys in the `TssMessageMap` have the following structure:
+
+```protobuf
+/**
+  * A key for use in the TssMessageMaps.
+  */
+message TssMessageMapKey {
+
+  /**
+   * A hash of the target roster for the associated value in the map.
+   */
+  bytes roster_hash = 1;
+
+  /**
+   * A number representing consensus order.<br/>
+   * This declares the order in which the mapped value came to consensus.
+   */
+  uint64 sequence_number = 2;
+}
+```
+
+The values in the `TssMessageMap` are of type `TssMessageTransaction` where the `target_roster_hash` in the value is
+the `roster_hash` in the key.
+
+Keys in the `TssVoteMap` have the following structure:
+
+```protobuf
+/**
+  * A key for use in the TssVoteMaps.
+  */
+message TssVoteMapKey {
+
+  /**
+   * A hash of the target roster for the associated value in the map.
+   */
+  bytes roster_hash = 1;
+
+  /**
+   * The node id of the node that created the TssVote.
+   */
+  uint64 node_id;
+}
+```
+
+The values in the `TssVoteMap` are of type `TssVoteTransaction` where the `target_roster_hash` in the value is the
+`roster_hash` in the key. The order in which votes come to consensus is not important. At least 1/2 of the weight
+must contribute yes votes that have the same vote structure for the candidate roster to be adopted. This ensures
+that at least 1 honest node has voted yes for the majority voting pattern.
+
+Lifecycle Invariants
+
+1. Inserts should only be made for candidate rosters that are going through the keying process.
+2. When a roster is no longer stored in the state, the corresponding key material and votes should be removed.
+3. Every roster that has become active after the TSS bootstrap process must have sufficient key material to be
+   able to recover the ledger id.
+
+### Core Behaviors
+
+We will add new behavior related to the bootstrap process of creating the ledger id
+and transferring the ability to sign messages from one roster to the next roster
+
+#### Startup for New Networks
+
+At the start of a new network there will be a new pre-genesis phase to generate the key material for the genesis
+roster and then restart the network, after adding the new key material and ledger id
+to the genesis state.
+
+The following startup sequence on new networks is modified from existing practices.
+
+1. The app will read the genesis address book and initial cryptography from disk
+2. The app will generate a genesis state and genesis roster.
+3. The app will ensure that it does not accept user transactions.
+4. The app will start the platform with the above and key the genesis roster with the TSS key material.
+5. After the genesis roster is keyed, the app will adopt the ledger id and restart the platform with the key
```

Comment:
There are a few changes for app here. Start and restart platform. Have these been discussed with services team?

Reply:
I think this _is_ that discussion 😄

Reply:
I missed that discussion. Thanks!